src/UNL/MediaHub.php

```diff
@@ -258,7 +258,7 @@ public static function checkMetadataProjection($url)
         }
 
         $projection = false;
-        $length = count($metadata);
+        $length = count($metadata->streams);
```

Comment:
Assumes $metadata->streams exists and is countable

Reply:
I believe that assumption is correct (assuming we start checking for ffprobe errors). When the application executes ffprobe, it does so with the `-show_streams` option:

```
ffmpeg/ffprobe -v quiet -print_format json -show_format -show_streams [file url]
```

 In the event no streams are returned, ffprobe will return an empty array, which `count()` can handle. The no results behavior can be demonstrated with the `-show_chapters` option when no chapters are present:

```
                "visual_impaired": 0,
                "clean_effects": 0,
                "attached_pic": 0,
                "timed_thumbnails": 0
            }
        }
    ],
    "chapters": [

    ],
    "format": {
        "filename": "/var/www/html/www/uploads/bfc9f750cd017fd88f74a8a7416bd68a.mp3",
        "nb_streams": 1,
```

The exception is if there's an error. Without the `-show-error` option (current code), the return is empty. If the option is enabled, then an invalid url will return:

```
{
    "error": {
        "code": -2,
        "string": "No such file or directory"
    }
}
```

Possible error handling:

```php
        $url = escapeshellarg($url);

-       exec(UNL_MediaHub::getFfprobePath() . " -v quiet -print_format json -show_format -show_streams $url", $return, $status);
+       exec(UNL_MediaHub::getFfprobePath() . " -v quiet -show_error -print_format json -show_format -show_streams $url", $return, $status);

        $json = "";

        $length = sizeof($return);
        for ($i=0; $i < $length; $i++) { 
            $json.=$return[$i];
        }

        $metadata = json_decode($json);
        
        if (!$metadata) {
            return false;
        }

+       if (isset($metadata->error)) {
+           throw new Exception('FFProbe error: ' . $metadata->error->string .' (Code ' . $metadata->error->code. ')');
+       }
+
        $projection = false;
        $length = count($metadata->streams);
        for ($i=0; $i < $length; $i++) { 
```

Reply:
I'm not sure if we can handle errors well enough to throw errors. 

I was thinking of a simple check like:

```
if (!$metadata || !isset($metadata->streams) || !is_countable($metadata->streams)) {
  return false;
}
```

Reply:
The requested change has been made. No ffprobe error handling is included. ffprobe will continue to fail silently.